### PR TITLE
ENH: Move warnings to error/__init__.py per GH27656

### PR DESCRIPTION
**Pull request by dataxerik**
- [x] xref #27656. this GitHub issue is being done in multiple parts
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

I noticed one of the strings didn't have a `Final` so I've added it. 

Besides the tslibs and OptionError errors, I think I've moved all the exceptions/warnings to the __errors.py. Is there anything else I should do?

### CODE REVIEW

File: pandas/errors/__init__.py [hunk 1]
```diff
@@ -469,10 +469,65 @@ class DatabaseError(OSError):
     """
 
 
+class PossiblePrecisionLoss(Warning):
+    """
+    Warning is raised when calling to_stata on a column where the column value is
```
Comment (mroeschke) [inline pandas/errors/__init__.py:474]: Could you ensure these summary sentences fit in one line? We are going to enable a validation to enforce this soon: https://github.com/pandas-dev/pandas/pull/47885

File: doc/source/whatsnew/v1.5.0.rst [hunk 2]
```diff
@@ -272,7 +272,7 @@ Other enhancements
 - A :class:`errors.PerformanceWarning` is now thrown when using ``string[pyarrow]`` dtype with methods that don't dispatch to ``pyarrow.compute`` methods (:issue:`42613`)
 - Added ``numeric_only`` argument to :meth:`Resampler.sum`, :meth:`Resampler.prod`, :meth:`Resampler.min`, :meth:`Resampler.max`, :meth:`Resampler.first`, and :meth:`Resampler.last` (:issue:`46442`)
 - ``times`` argument in :class:`.ExponentialMovingWindow` now accepts ``np.timedelta64`` (:issue:`47003`)
-- :class:`.DataError`, :class:`.SpecificationError`, :class:`.SettingWithCopyError`, :class:`.SettingWithCopyWarning`, :class:`.NumExprClobberingError`, :class:`.UndefinedVariableError`, and :class:`.IndexingError` are now exposed in ``pandas.errors`` (:issue:`27656`)
+- Various errors and warnings are now exposed in ``pandas.errors`` (:issue:`27656`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.5.0.rst:275]: I think it's still good to list the exact errors & warnings you exposed here

**Comment by mroeschke** [comment]
In a follow up PR, you could maybe add a pre-commit hook to validate that errors are only defined in `pandas/_libs` or `pandas/errors/__init__.py` to close out the issue

File: pandas/errors/__init__.py (hunk 1 above)
Comment (dataxerik) [reply pandas/errors/__init__.py]: Gotcha, updated

File: doc/source/whatsnew/v1.5.0.rst (hunk 2 above)
Comment (dataxerik) [reply doc/source/whatsnew/v1.5.0.rst]: Updated

**Comment by dataxerik** [comment]
> In a follow up PR, you could maybe add a pre-commit hook to validate that errors are only defined in `pandas/_libs` or `pandas/errors/__init__.py` to close out the issue

Gotcha, I'll work on that next

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks again @dataxerik!

**Pull request merged**